### PR TITLE
Remove requester-specific parameters from templates

### PR DESCRIPTION
**Pull request by bbenligiray**
We removed short requests (https://github.com/api3dao/airnode/pull/220). Templates are composed of requester-agnostic (`providerId`, `endpointId`, `parameters`) and requester-specific parameters (`requesterIndex`, `designatedWallet`, `fulfillAddress`, `fulfillFunctionId`). Only short requests used the requester-specific parameters in the templates, and they became redundant with the removal of short requests. This PR removes the requester-specific parameters from templates.

Also, provider initialization steps have been reordered to verify API calls before fetching templates. because we know the requester indexes/designated wallets as soon as we receive the requests.

### CODE REVIEW

File: packages/node/src/evm/handlers/initialize-provider.test.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ describe('initializeProvider', () => {
         requestCount: '2',
         requesterIndex: '2',
         status: 'Pending',
-        templateId: '0xe315dcd8305800ebdf4c188fa85c602387d36df23de6927d28820d695a3c0deb',
+        templateId: '0x537522dde3d2d4fb409b9c278d0b08de85f5aea7339efd23c5a8d7316242d2db',
```
Comment (bbenligiray) [inline packages/node/src/evm/handlers/initialize-provider.test.ts:103]: Confirmed for
```js
const providerId = '0x19255a4ec31e89cea54d1f125db7536e874ab4a96b4d4f6438668b6bb10a6adb';
const endpointId = '0xac2e948e29db14b568a3cbaeedc66c0f9b5c5312f6b562784889e8cbd6a6dd9e';
const parameters = [
	{ "type": "bytes32", "name": "to", "value": "USD" },
	{ "type": "bytes32", "name": "_type", "value": "int256" },
	{ "type": "bytes32", "name": "_path", "value": "result" },
	{ "type": "bytes32", "name": "_times", "value": "100000" }
      ];
```

File: packages/node/src/evm/handlers/initialize-provider.test.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ describe('initializeProvider', () => {
         endpointId: '0xac2e948e29db14b568a3cbaeedc66c0f9b5c5312f6b562784889e8cbd6a6dd9e',
         fulfillAddress: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
         fulfillFunctionId: '0x48a4157c',
-        id: '0x7073d6a5530629274041f7766f4c3b94118c8cd2932c8af27b166e1c3cd94e30',
+        id: '0xe2347d69997f674902398ea1b43239ed8f4fcf81a1baa7a904e37f15e0c85e7e',
```
Comment (bbenligiray) [inline packages/node/src/evm/handlers/initialize-provider.test.ts:85]: Confirmed for
```js
const noRequests = '2';
const clientAddress = '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0';
const templateId = '0x537522dde3d2d4fb409b9c278d0b08de85f5aea7339efd23c5a8d7316242d2db';
const parameters = [
	{ "type": "bytes32", "name": "from", "value": "ETH" }
      ];
const encodedParameters = airnodeAbi.encode(parameters);
```

File: packages/node/src/evm/templates/template-verification.test.ts [hunk 3]
```diff
@@ -5,20 +5,12 @@ import { RequestErrorCode, RequestStatus } from 'src/types';
...
-      'fulfillFunctionId',
-      'encodedParameters',
-    ]);
+    expect(verification.TEMPLATE_VALIDATION_FIELDS).toEqual(['providerId', 'endpointId', 'encodedParameters']);
   });
 });
 
 describe('verify', () => {
-  const TEMPLATE_ID = '0x101c36202b92b358adda664515ea920f76b1edcf5f1285ab0c845f54638b45aa';
+  const TEMPLATE_ID = '0xe29a81893520cc4964bea1bc003e836e658c8043ba841fb7e5f7f91fe99fbb5b';
```
Comment (bbenligiray) [inline packages/node/src/evm/templates/template-verification.test.ts:13]: Confirmed for
```js
const providerId = '0x9e5a89de5a7e780b9eb5a61425a3a656f0c891ac4c56c07037d257724af490c9';
const endpointId = '0x2f3a3adf6daf5a3bb00ab83aa82262a6a84b59b0a89222386135330a1819ab48';
const encodedParameters = '0x6466726f6d63455448';
```

File: packages/node/src/evm/verification/api-call-verification.test.ts [hunk 4]
```diff
@@ -24,9 +24,9 @@ describe('verifyApiCallIds', () => {
     const apiCall = fixtures.requests.createApiCall({
       clientAddress: '0x7f7d1Aa0792aC39f43C6e7FA2ec31258Fc5FD612',
       encodedParameters: '0x62746f6355534466616d6f756e746131',
-      id: '0x8dc1d7fe6e5d68f429e536c7c7efedcc081b142bb0ad5fced130a33a12f2aa46',
+      id: '0x564ea6249a5ade738e037320153a7052043f5268afcc5340f9d2b8f4cc29ccc1',
```
Comment (bbenligiray) [inline packages/node/src/evm/verification/api-call-verification.test.ts:27]: Confirmed for
```js
const noRequests = '0';
const clientAddress = '0x7f7d1Aa0792aC39f43C6e7FA2ec31258Fc5FD612';
const templateId = '0xe29a81893520cc4964bea1bc003e836e658c8043ba841fb7e5f7f91fe99fbb5b';
const encodedParameters = '0x62746f6355534466616d6f756e746131';
```

File: packages/node/src/types.ts [hunk 5]
```diff
@@ -67,22 +67,18 @@ export interface ApiCall {
   readonly fulfillAddress: string | null;
   readonly fulfillFunctionId: string | null;
   readonly parameters: ApiCallParameters;
-  readonly providerId: string;
+  readonly providerId: string | null;
```
Comment (bbenligiray) [inline packages/node/src/types.ts:70]: Here, `providerId` should be treated the same as `endpointId` (full requests get it from request parameters, regular requests get it from the template)

File: packages/node/test/fixtures/evm/event-logs.ts [hunk 6]
```diff
@@ -13,11 +13,11 @@ export function buildClientRequest(overrides?: Partial<Log>): Log {
...
+      '0x00000000000000000000000000000000000000000000000000000000000000020000000000000000000000009fe46736679d2d9a65f0992f2272de9f3c7fa6e0537522dde3d2d4fb409b9c278d0b08de85f5aea7339efd23c5a8d7316242d2db0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000a46c4b41d72ada9d14157b28a8a2db97560fff120000000000000000000000009fe46736679d2d9a65f0992f2272de9f3c7fa6e048a4157c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000060316200000000000000000000000000000000000000000000000000000000000066726f6d000000000000000000000000000000000000000000000000000000004554480000000000000000000000000000000000000000000000000000000000',
```
Comment (bbenligiray) [inline packages/node/test/fixtures/evm/event-logs.ts:16]: I just found and replaced the `templateId` here. To what you want with this information.

File: packages/node/src/evm/templates/template-application.ts [hunk 7]
```diff
@@ -22,12 +22,9 @@ function applyTemplate(
 ): ClientRequest<ApiCall> {
   return {
     ...request,
-    // NOTE: template attributes can be overwritten by the request attributes
-    designatedWallet: request.designatedWallet || template.designatedWallet,
+    // providerId and endpointId will either be request or a template attribute
+    providerId: request.providerId || template.providerId,
```
Comment (bbenligiray) [inline packages/node/src/evm/templates/template-application.ts:26]: I'm not sure why this wasn't applying the `providerId` before

File: packages/node/src/evm/handlers/initialize-provider.ts [hunk 8]
```diff
@@ -90,15 +90,28 @@ export async function initializeProvider(
   const state3 = state.update(state2, { requests: groupedRequests });
 
   // =================================================================
-  // STEP 4: Fetch and apply templates to API calls
+  // STEP 4: Validate API calls
```
Comment (bbenligiray) [inline packages/node/src/evm/handlers/initialize-provider.ts:93]: Check this file carefully

File: packages/node/src/evm/templates/template-application.ts (hunk 7 above)
Comment (andreogle) [reply packages/node/src/evm/templates/template-application.ts:26]: Weird. I took a quick look and can't find anything else that would have applied it for short/regular requests.

File: packages/node/src/evm/verification/api-call-verification.test.ts (hunk 4 above)
Comment (andreogle) [reply packages/node/src/evm/verification/api-call-verification.test.ts:27]: Small thing, but `noRequests` (`requestCount`) will always be >= 1 now. I'm not sure that it's very important for the unit tests though

File: packages/node/src/types.ts (hunk 5 above)
Comment (andreogle) [reply packages/node/src/types.ts:70]: Good point 👍🏻

**Review by andreogle — APPROVED** [review]
Looks great 👍🏻

File: packages/node/src/evm/verification/api-call-verification.test.ts (hunk 4 above)
Comment (bbenligiray) [reply packages/node/src/evm/verification/api-call-verification.test.ts:27]: It is easy enough to change, I updated it to `5`

**Pull request merged**